### PR TITLE
BDOG-3038 add more query params to deleted repos API + handle repos b…

### PR DESCRIPTION
**Pull request by Shnick**
…eing recreated via scheduler

### CODE REVIEW

File: app/uk/gov/hmrc/teamsandrepositories/persistence/DeletedRepositoriesPersistence.scala [hunk 1]
```diff
@@ -34,32 +32,55 @@ class DeletedRepositoriesPersistence @Inject()(
...
-    ).flatten
-
+  // Remove when repo has been recreated - added to repositories collection
+  def deleteRepos(repoNames: Seq[String]): Future[Long] =
     collection
-      .find(if (filters.isEmpty) BsonDocument() else Filters.and(filters: _*))
-      .collation(if (name.isDefined) Collations.caseInsensitive else Collation.builder().build())
+      .deleteMany(
+        Filters.or(repoNames.map(name => Filters.eq("name", name)): _*)
+      , DeleteOptions().collation(Collations.caseInsensitive)
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/teamsandrepositories/persistence/DeletedRepositoriesPersistence.scala:81]: does this only apply when repoNames is non empty?

File: test/uk/gov/hmrc/teamsandrepositories/controller/DeletedRepositoriesControllerSpec.scala [hunk 2]
```diff
@@ -94,17 +95,39 @@ class DeletedRepositoriesControllerSpec
...
       status(result) mustBe OK
       contentAsJson(result) mustBe Json.toJson(expectedModel)
 
-      verify(mockDeletedRepositoriesPersistence).find(eqTo(Some("Foo")), eqTo(None))
+      verify(mockDeletedRepositoriesPersistence).find(eqTo(Some("Foo")), eqTo(None), eqTo(None), eqTo(None))
     }
+
+    "get all deleted services" in {
+
+      implicit val reads: OFormat[DeletedGitRepository] = DeletedGitRepository.apiFormat
```
Comment (colin-lamed) [inline test/uk/gov/hmrc/teamsandrepositories/controller/DeletedRepositoriesControllerSpec.scala:113]: these tests should be asserting expected json, to ensure the internal apiFormat is correct

**Review by colin-lamed — APPROVED** [review]

File: app/uk/gov/hmrc/teamsandrepositories/persistence/DeletedRepositoriesPersistence.scala (hunk 1 above)
Comment (Shnick) [reply app/uk/gov/hmrc/teamsandrepositories/persistence/DeletedRepositoriesPersistence.scala]: added a test for empty

File: test/uk/gov/hmrc/teamsandrepositories/controller/DeletedRepositoriesControllerSpec.scala (hunk 2 above)
Comment (Shnick) [reply test/uk/gov/hmrc/teamsandrepositories/controller/DeletedRepositoriesControllerSpec.scala]: fixed

**Review by colin-lamed — APPROVED** [review]

**Pull request merged**